src/main/java/org/janelia/saalfeldlab/paintera/ui/opendialog/meta/MetaPanel.java

```diff
@@ -182,12 +187,21 @@ public MetaPanel() {
 	final GridPane rawMinMax = new GridPane();
 	rawMinMax.getColumnConstraints().add(cc);
 	rawMinMax.add(new Label("Intensity Range"), 0, 0);
-	rawMinMax.add(this.min, 1, 0);
-	rawMinMax.add(this.max, 2, 0);
-	this.min.setPromptText("min");
-	this.max.setPromptText("max");
-	this.min.setPrefWidth(TEXTFIELD_WIDTH);
-	this.max.setPrefWidth(TEXTFIELD_WIDTH);
+	rawMinMax.add(this.min.getTextField(), 1, 0);
+	rawMinMax.add(this.max.getTextField(), 2, 0);
+
+	/* restrict to decimal numbers only */
+	final EventHandler<KeyEvent> decimalCharFilter = event -> {
+	  if (!".0123456789".contains(event.getCharacter())) {
+		event.consume();
+	  }
+	};
+	this.min.getTextField().addEventFilter(KeyEvent.KEY_TYPED, decimalCharFilter);
+	this.max.getTextField().addEventFilter(KeyEvent.KEY_TYPED, decimalCharFilter);
+	this.min.getTextField().setPromptText("min");
+	this.max.getTextField().setPromptText("max");
+	this.min.getTextField().setPrefWidth(TEXTFIELD_WIDTH);
+	this.max.getTextField().setPrefWidth(TEXTFIELD_WIDTH);
```

Comment:
Fix for intensity setting not working on source import (https://github.com/saalfeldlab/paintera/issues/300)